src/main/java/seedu/address/model/person/Address.java

```diff
@@ -9,27 +9,29 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.logic.parser.exceptions.ParseException;
 
 /**
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
-
     public static final String MESSAGE_CONSTRAINTS = "Addresses can take any values, and it should not be blank\n"
             + LABEL_MESSAGE
             + "\n\n"
             + "Multiple addresses are allowed but most adhere to the following conditions: \n"
             + "1. For 1 address only, the label is optional so: ADDRESS or ADDRESS (LABEL).\n"
             + "2. For multiple emails, the label is compulsory so: ADDRESS1 (LABEL1) ADDRESS2 (LABEL2) ... "
             + "ADDRESSN (LABELN).";
-
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
     public static final String ADDRESS_VALIDATION_REGEX = "[^\\s].*";
-
+    private static final Logger logger = LogsCenter.getLogger(Email.class);
```

Comment:
```suggestion
    private static final Logger logger = LogsCenter.getLogger(Address.class);
```
Perhaps this should be `Address.class` instead of `Email.class`?